src/inmanta/ast/statements/generator.py

```diff
@@ -888,7 +891,7 @@ def normalize(self, *, lhs_attribute: Optional[AttributeAssignmentLHS] = None) -
             if not self.wrapped_kwargs and (self._lhs_attribute is None or len(self._required_dynamic_args) > 1):
                 raise IndexAttributeMissingInConstructorException(self, self.type, self._required_dynamic_args)
 
-        self._normalize_rhs(inindex)
+        self.anchors.extend(self._normalize_rhs(inindex))
```

Comment:
@sanderr for my test to pass (and the anchormap to contain the anchor to the constructor) I had make change this. Do you think this is an issue or this could be done in a better way? 

(I still need to move the anchor logic to the normalizers for the other cases)

Reply:
Why not just `append` to the anchors from that method?

Reply:
this is indeed way better, I don't know why I didn't do it like that immediately